controllers/certificatesigningrequests_controller.go

```diff
@@ -102,8 +105,34 @@ func (r *certificateSigningRequestsReconciler) Reconcile(ctx context.Context,
 	return ctrl.Result{}, r.reconcileCSR(certificateSigningRequest)
 }
 
-// reconcileCSR handles the CSR validation and approval
+// reconcileCSR kicks off CSR validation and approval. In case of update conflicts, the approval process is wrapped
+// with retry logic using the Update --> Get --> Retry Update pattern. Retry will occur until specified timeout
 func (r *certificateSigningRequestsReconciler) reconcileCSR(request *certificates.CertificateSigningRequest) error {
+	err := wait.Poll(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
+		err := r.validateAndApprove(request)
+		if err != nil {
```

Comment:
This is causing a large portion of this function be indented. Can the inverse be done?
```
if err == nil {
    return true, nil
}

			if k8sapierrors.IsConflict(err) {
...
```